### PR TITLE
R2s script add tests

### PR DESCRIPTION
**Pull request by zxkjack123**
This PR adds some R2S examples for voxel, sub-voxel and unstructured mesh mode. These example files include both input files and expected output files.
Test functions for r2s step1 and step2 of these three modes are added as well.

### CODE REVIEW

File: pyne/r2s.py [hunk 1]
```diff
@@ -101,6 +102,8 @@ def irradiation_setup(flux_mesh, cell_mats, cell_fracs, alara_params, tally_num=
         The if False, geometry discretization will be done with randomly fired
         rays. If true, a grid of sqrt(num_rays) x sqrt(num_rays) rays is used
         for each mesh row.
+    cell_fracs : record array
```
Comment (gonuke) [inline pyne/r2s.py]: This appears above in the comments

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks @zxkjack123 - this will be useful!

I don’t think we need any of the nuclear data files used by ALARA to be included in this PR.

**Comment by zxkjack123** [comment]
@gonuke That's true. Nuclear data are for ALARA, I have removed that.

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
Thanks @zxkjack123

**Pull request merged**